addons/hr_evaluation/hr_evaluation.py

```diff
@@ -1,375 +1,465 @@
-            partner_id = user_obj.read(cr, uid, vals.get('user_id'), fields=['partner_id'], context=context)['partner_id'][0]
-        else:
-            partner_id = None
-
-        user_input_obj = self.pool.get('survey.user_input')
-
-        if not vals.get('deadline'):
-            vals['deadline'] = (datetime.now() + timedelta(days=28)).strftime(DF)
-
-        ret = user_input_obj.create(cr, uid, {'survey_id': survey_id,
-                                              'deadline': vals.get('deadline'),
-                                              'type': 'link',
-                                              'partner_id': partner_id}, context=context)
-        vals['request_id'] = ret
-        return super(hr_evaluation_interview, self).create(cr, uid, vals, context=context)
-
-    def name_get(self, cr, uid, ids, context=None):
-        if not ids:
-            return []
-        reads = self.browse(cr, uid, ids, context=context)
-        res = []
-        for record in reads:
-            name = record.survey_id.title
-            res.append((record['id'], name))
+    _description = "Employee Appraisal"
+    _order = 'date_close, interview_deadline'
+
+    EVALUATION_STATE = [
+        ('new', 'To Start'),
+        ('pending', 'Appraisal Sent'),
+        ('done', 'Done')
+    ]
+
+    @api.model
+    def _set_default_template(self):
+        return self.env.ref('hr_evaluation.email_template_appraisal')
+
+    @api.one
+    def _get_total_appraisal(self):
+        return self.env['survey.user_input'].search([
+            ('survey_id', 'in', [self.appraisal_manager_survey_id.id, self.appraisal_colleagues_survey_id.id, self.appraisal_self_survey_id.id, self.appraisal_subordinates_survey_id.id]),
+            ('type', '=', 'link'), ('evaluation_id', '=', self.id)])
+
+    @api.one
+    @api.depends('state', 'interview_deadline')
+    def _get_tot_sent_appraisal(self):
+        for records in self._get_total_appraisal():
+            self.tot_sent_appraisal = len(records)
+
+    @api.one
+    def _get_tot_completed_appraisal(self):
+        for records in self._get_total_appraisal():
+            self.tot_comp_appraisal = len(records.filtered(lambda r: r.state == 'done'))
+
+    meeting_id = fields.Many2one('calendar.event', string='Meeting')
+    interview_deadline = fields.Date(string="Final Interview", select=True)
+    employee_id = fields.Many2one('hr.employee', required=True, string='Employee')
+    department_id = fields.Many2one('hr.department', string='Department')
+    evaluation = fields.Text(string='Evaluation Summary', help="If the evaluation does not meet the expectations, you can propose an action plan")
+    state = fields.Selection(EVALUATION_STATE, string='Status', track_visibility='onchange', required=True, readonly=True, copy=False, default='new', select=True)
+    date_close = fields.Datetime(string='Appraisal Deadline', select=True, required=True)
+    appraisal_manager = fields.Boolean(string='Manager')
```

Comment:
What does it mean ? The employee is a manager, the apparaisal will be a manager, ... ? I suppose it is a field telling "Appraisal from the Manager" ?